config/config.go

```diff
@@ -1657,17 +1657,23 @@ func (c *Config) SaveConfigToFile(configPath string) error {
 	if err != nil {
 		return err
 	}
-	writer, err := file.Writer(defaultPath)
-	if err != nil {
-		return err
+	var writer *os.File
+	provider := func() (io.Writer, error) {
+		writer, err = file.Writer(defaultPath)
```

Comment:
Just a suggestion but we could return an io.WriteCloser and call defer close after the error check from the calling fuction.

Reply:
mm... I'm not convinced... `bytes.Buffer` for instance doesn't have the `Close()` implemented. I think leaving it as is is more generic.

Reply:
That's a really good point! My mistake. Thanks.